core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4548,29 +4548,42 @@ public Object getTarget() {
         try {
             checkPermission(READ);
         } catch (AccessDeniedException e) {
-            String rest = Stapler.getCurrentRequest().getRestOfPath();
-            for (String name : ALWAYS_READABLE_PATHS) {
-                if (rest.startsWith(name)) {
-                    return this;
-                }
-            }
-            for (String name : getUnprotectedRootActions()) {
-                if (rest.startsWith("/" + name + "/") || rest.equals("/" + name)) {
-                    return this;
-                }
-            }
-
-            // TODO SlaveComputer.doSlaveAgentJnlp; there should be an annotation to request unprotected access
-            if (rest.matches("/computer/[^/]+/slave-agent[.]jnlp")
-                && "true".equals(Stapler.getCurrentRequest().getParameter("encrypt"))) {
+            if (isPathUnprotected(Stapler.getCurrentRequest().getRestOfPath())) {
                 return this;
             }
 
-
             throw e;
         }
         return this;
     }
+    
+    /**
+     * Test a path to see if it does not require authentication
```

Comment:
Yes, that works; I wasn't seeing the name and thinking "switch the logic to match that name" before, hence my confusion. Probably just because I have been looking at this late at night after long days.